src/main/groovy/edu/oregonstate/mist/inventory/resources/InventoryResource.groovy

```diff
@@ -89,22 +90,41 @@ class InventoryResource extends Resource {
         List<Error> errors = getErrors(newResultObject)
 
         if (errors) {
-            Response.ResponseBuilder responseBuilder = Response.status(Response.Status.BAD_REQUEST)
-            return responseBuilder.entity(errors).build()
+            return errorsResponse(errors)
         }
         Inventory newInventory = resultObjectToInventory(newResultObject)
         newInventory.id = newResultObject.data['id'] ?: randomUUID() as String
 
         inventoryDAOWrapper.createInventory(newInventory)
 
-        ResourceObject newInventoryFromDB = inventoryDAOWrapper.getInventoryById(newInventory.id)
+        ResourceObject inventoryFromDB = inventoryDAOWrapper.getInventoryById(newInventory.id)
 
-        if (!newInventoryFromDB) {
-            internalServerError(ErrorMessages.getInventoryError()).build()
+        if (!inventoryFromDB) {
+            getInventoryError()
         } else {
-            created(new ResultObject(
-                    data: newInventoryFromDB
-            )).build()
+            created(addSingleObjectToResultObject(inventoryFromDB)).build()
+        }
+    }
+
+    @Timed
+    @PUT
+    @Path('{id: [0-9a-zA-Z-]+}')
+    Response updateInventory(@PathParam('id') String inventoryID,
+                             @Valid ResultObject resultObject) {
+        List<Error> errors = getErrors(resultObject, inventoryID)
+
+        if (errors) {
+            return errorsResponse(errors)
+        }
+
+        inventoryDAOWrapper.updateInventory(resultObjectToInventory(resultObject), inventoryID)
+
+        ResourceObject inventoryFromDB = inventoryDAOWrapper.getInventoryById(inventoryID)
```

Comment:
How would the user know if there was a problem updating some of the inventory json object? For example, what if the datasource failed to save, but the rest of the json object saves correctly? Would there be any errors displayed to the end user?